packages/kit/src/runtime/client/client.js

```diff
@@ -2120,6 +2120,16 @@ function _start_router() {
 
 		event.preventDefault();
 
+		// allow the browser to repaint before navigating —
+		// this prevents INP scores being penalised
+		new Promise((fulfil) => {
```

Comment:
I'm assuming you need to await this promise?

Reply:
that would help!

```suggestion
		await new Promise((fulfil) => {
```